website/content/docs/lambda/invoke-from-lambda.mdx

```diff
@@ -0,0 +1,301 @@
+    ```
+    </CodeBlockConfig>
+
+1. Configure and deploy the Lambda function. Refer to the [Lambda function configuration](#lambda-function-configuration) reference for information about all available options. There are several methods for deploying Lambda functions. The following example uses Terraform to deploy a function that can invoke the `static-server` upstream service using mTLS data stored under the `/lambda_extension_data` prefix:
+
+    <CodeBlockConfig filename="lambda-function.tf">
+
+    ```hcl
+    resource "aws_lambda_function" "example" {
+    …
+    function_name = "lambda"
+    role = aws_iam_role.lambda.arn
+    tags = {
+        "serverless.consul.hashicorp.com/v1alpha1/lambda/enabled" = "true"
+    }
+    variables = {
+        environment = {
+        CONSUL_MESH_GATEWAY_URI = var.mesh_gateway_http_addr
+        CONSUL_SERVICE_UPSTREAMS = "static-server:2345:dc1"
+        CONSUL_EXTENSION_DATA_PREFIX = “/lambda_extension_data”
+        }
+    }
+    layers = [aws_lambda_layer_version.consul_lambda_extension.arn]  
+    ```
+
+    </CodeBlockConfig>
+
+1. Issue the `terraform apply` command and Consul automatically configures a service for the Lambda function.
+
+
+### Lambda function configuration
+
+Define  the following environment variables to configure each Lambda function. The configurations apply to each Lambda function in your environment:
+
+| Variable | Description | Default |
+| ---      | ---         | ---     |
+| `CONSUL_MESH_GATEWAY_URI` | Specifies the URI where the mesh gateways that the plugin makes requests are running. The mesh gateway should be registered in the same Consul datacenter and partition that the service is running in. For optimal performance, this mesh gateway should run in the same AWS region. | none |
+| <nobr>`CONSUL_EXTENSION_DATA_PREFIX`</nobr> | Specifies the prefix that the plugin pulls configuration data from. The data must be located in the following directory: <br/>`“${CONSUL_EXTENSION_DATA_PREFIX}/${CONSUL_SERVICE_PARTITION}/${CONSUL_SERVICE_NAMESPACE}/<lambda-function-name>”` | none |
+| `CONSUL_SERVICE_NAMESPACE` | Specifies the Consul namespace the service is registered into. | `default` |
+| `CONSUL_SERVICE_PARTITION` | Specifies the Consul partition the service is registered into. | `default` |
+| `CONSUL_REFRESH_FREQUENCY` | Specifies the amount of time the extension waits before re-pulling data from the Parameter Store. Use [Go `time.Duration`](https://pkg.go.dev/time@go1.19.1#ParseDuration) string values, for example, `”30s”`. <br/>The time is added to the duration configured in the Lambda registrator `sync_frequency_in_minutes` configuration. Refer to [Lambda registrator configuration options](/docs/lambda/registration/automate#lambda-registrator-configuration-options). The combined configurations determine how stale the data may become. Lambda functions can run for up to 14 hours. We recommend configuring an acceptable value to preview stale certificates. | `“5m”` |
```

Comment:
What does "We recommend configuring an acceptable value to preview stale certificates" mean?